timer/src/main/java/com/toggl/timer/di/TimerModule.kt

```diff
@@ -91,26 +94,42 @@ class TimerModule {
     @Provides
     @Singleton
     internal fun timerReducer(
+        timeEntryReducer: TimeEntryReducer,
         timeEntriesLogReducer: TimeEntriesLogReducer,
         startEditReducer: StartEditReducer,
         runningTimeEntryReducer: RunningTimeEntryReducer,
         projectReducer: ProjectReducer
     ): TimerReducer {
 
         return combine(
-            timeEntriesLogReducer.pullback(
+            timeEntriesLogReducer.decorateWith(timeEntryReducer,
+                mapToLocalState = { TimeEntryState(it.timeEntries) },
+                mapToLocalAction = { if (it !is TimeEntriesLogAction.TimeEntryHandling) null else it.timeEntryAction },
```

Comment:
it reads a tiny better like this imo:
```suggestion
                mapToLocalAction = { (it as? TimeEntriesLogAction.TimeEntryHandling)?.timeEntryAction },
```